### ISSUE TITLE
libpulsar Homebrew formula fails

### ISSUE THREAD

**Issue by Jim-Lambert-Bose** [apache/pulsar#4286]
libpulsar Homebrew formula fails to install and reports that the "protobuf@2.6" dependency is "missing"

**To Reproduce**
Following the instructions from: https://pulsar.apache.org/docs/latest/clients/Cpp/#MacOS-684t0p

Running the brew command specified:

```
$ brew install https://raw.githubusercontent.com/apache/incubator-pulsar/master/pulsar-client-cpp/homebrew/libpulsar.rb

Updating Homebrew...
######################################################################## 100.0%
Error: No available formula with the name "protobuf@2.6" (dependency of libpulsar)
==> Searching for a previously deleted formula (in the last month)...
Warning: homebrew/core is shallow clone. To get complete history run:
  git -C "$(brew --repo homebrew/core)" fetch --unshallow

Error: No previously deleted formula found.
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
==> Searching taps...
==> Searching taps on GitHub...
Error: No formulae found in taps.
```

**Desktop (please complete the following information):**
MacOS Mojave version 10.14.4

**Comment by merlimat** [comment]
@Jim-Lambert-Bose the doc page above is from the old site and it's outdated (we need to get that out the google index). 
Please refer to https://pulsar.apache.org/docs/en/client-libraries-cpp 

To install on Mac, it would be just `brew install libpulsar` since it's already included in Homebrew-Core

**Comment by merlimat** [comment]
BTW: just removed the outdated docs from site.

**Comment by Jim-Lambert-Bose** [comment]
@merlimat that's 2.3.1 but that won't work with the go-client correct?
https://github.com/apache/pulsar/issues/4055

**Comment by merlimat** [comment]
You just need to make sure to pick the `v2.3.1` tag of the Go code

**Comment by merlimat** [comment]
Ok I haven’t fully read the other issue.... not sure what’s the problem there

**Comment by Jim-Lambert-Bose** [comment]
There is no v2.3.x tags on the "go code" in the pulsar repo.  How in the world does that even happen 🤷‍♂ 

I was able to find the git hash of the v2.3.1 commit and pull the right module version via:
`go get github.com/apache/pulsar/pulsar-client-go@ 7a6f2cca0fdb01e4a1ea1c72d907217a13af8ff1`

But... in the end, there is some compatibility issue.  Even the most basic go example client won't build:
```
package main
import (
	"log"
	"runtime"

	"github.com/apache/pulsar/pulsar-client-go/pulsar"
)
func main() {
	client, err := pulsar.NewClient(pulsar.ClientOptions{
		URL:                     "pulsar://localhost:6650",
		OperationTimeoutSeconds: 5,
		MessageListenerThreads:  runtime.NumCPU(),
	})

	if err != nil {
		log.Fatalf("Could not instantiate Pulsar client: %v", err)
	}
	client.Close()
}
```

```
c_message.go line #91
cannot use size (type _Ctype_int) as type _Ctype_ulong in assignment
```

**Comment by merlimat** [comment]
> There are no v2.3.x tags on the "go code" in the pulsar repo. How in the world does that even happen 🤷‍♂

The tag is set on the repo: https://github.com/apache/pulsar/tree/v2.3.1
Go client code is in the `pulsar-client-go` directory in the repo: https://github.com/apache/pulsar/tree/v2.3.1/pulsar-client-go

**Comment by Jim-Lambert-Bose** [comment]
Pulling the v2.3.1 does not work:
```
$ go get github.com/apache/pulsar/pulsar-client-go@v2.3.1                                  
go: finding github.com/apache/pulsar/pulsar-client-go v2.3.1
go: finding github.com/apache/pulsar v2.3.1
go get github.com/apache/pulsar/pulsar-client-go@v2.3.1: unknown revision pulsar-client-go/v2.3.1
```

But, I got it work via...
```
# latest commit in the branch 2.3 branch
# https://github.com/apache/pulsar/commits/branch-2.3 
# which is: 1f4a836a46483db408e0bf39903215974262d946

# install 2.3.1
 brew install libpulsar  

# get the go-client module
go get github.com/apache/pulsar/pulsar-client-go@1f4a836a46483db408e0bf39903215974262d946

$ go run main.go
2019/05/16 13:21:59.388 c_client.go:68: [info] INFO  | ClientImpl:492 | Closing Pulsar client
```

**Comment by Jim-Lambert-Bose** [comment]
I updated https://github.com/apache/pulsar/issues/4055

**Comment by merlimat** [comment]
Got it. The problem is that 2.3.1 was missing a fix (#3759) that went in master and was backported in `branch-2.3`. This will be included in the upcoming 2.3.2 which will be released shortly.
